src/main/java/sirius/biz/jobs/batch/file/DictionaryBasedImport.java

```diff
@@ -0,0 +1,158 @@
+
+import sirius.biz.importer.format.ImportDictionary;
+import sirius.biz.jobs.params.BooleanParameter;
+import sirius.biz.jobs.params.Parameter;
+import sirius.biz.process.ProcessContext;
+import sirius.biz.process.logs.ProcessLog;
+import sirius.kernel.async.TaskContext;
+import sirius.kernel.commons.Callback;
+import sirius.kernel.commons.Context;
+import sirius.kernel.commons.Strings;
+import sirius.kernel.commons.Tuple;
+import sirius.kernel.commons.Values;
+import sirius.kernel.commons.Watch;
+import sirius.kernel.health.Exceptions;
+import sirius.kernel.health.HandledException;
+import sirius.kernel.health.Log;
+import sirius.kernel.nls.NLS;
+
+/**
+ * Provides a helper to process line based values using a {@link ImportDictionary}.
+ * <p>
+ * This essentially maps a {@link Values} to a {@link Context} for each row.
+ */
+public class DictionaryBasedImport {
+
+    private static final String MESSAGE_KEY = "message";
+    private static final String ERROR_IN_ROW_KEY = "LineBasedJob.errorInRow";
+
+    /**
+     * Contains the parameter which is used to determine if empty values should be ignored.
+     */
+    public static final Parameter<Boolean> IGNORE_EMPTY_PARAMETER =
+            new BooleanParameter("ignoreEmpty", "$DictionaryBasedImportJobFactory.ignoreEmpty").withDescription(
+                    "$DictionaryBasedImportJobFactory.ignoreEmpty.help").build();
+
+    protected final ProcessContext process;
+    protected final boolean ignoreEmptyValues;
+    protected final ImportDictionary dictionary;
+    protected Callback<Tuple<Integer, Context>> rowHandler;
+
+    /**
+     * Creates a new job for the given factory, name and process.
+     *
+     * @param dictionary the import dictionary to use
+     * @param process    the process context itself
+     */
+    protected DictionaryBasedImport(ImportDictionary dictionary,
+                                    ProcessContext process,
+                                    boolean ignoreEmptyValues,
+                                    Callback<Tuple<Integer, Context>> rowHandler) {
+
+        this.dictionary = dictionary;
+        this.process = process;
+        this.ignoreEmptyValues = ignoreEmptyValues;
+        this.rowHandler = rowHandler;
+    }
+
+    /**
+     * Invoked for each row in an import file.
+     *
+     * @param index the line number to process
+     * @param row   the data to process
+     */
+    public void handleRow(int index, Values row) {
+        if (index == 1) {
+            handleHeaderRow(row);
+        } else {
+            handleDataRow(index, row);
+        }
+    }
+
+    private void handleHeaderRow(Values row) {
+        if (dictionary.hasIdentityMapping()) {
+            boolean problemsFound = dictionary.detectHeaderProblems(row, (problem, isFatal) -> {
+                if (Boolean.TRUE.equals(isFatal)) {
+                    process.log(ProcessLog.error()
+                                          .withNLSKey(ERROR_IN_ROW_KEY)
```

Comment:
Would be nice to enhance these messages with the current file name. In the process logs one sees: "Error in row #: the error...". But since the jobs can process several files, would be a good hint to show Error in file 'foo', row #: the error... "